Code/Tools/TestImpactFramework/Runtime/Code/Source/TestImpactUtils.cpp

```diff
@@ -243,4 +243,256 @@ namespace TestImpact
             throw(Exception(AZStd::string::format("Unexpected client test case result: %u", aznumeric_cast<AZ::u32>(result))));
         }
     }
+
+    SuiteType SuiteTypeFromString(const AZStd::string& suiteType)
+    {
+        if (suiteType == SuiteTypeAsString(SuiteType::Main))
+        {
+            return SuiteType::Main;
+        }
+        else if (suiteType == SuiteTypeAsString(SuiteType::Periodic))
+        {
+            return SuiteType::Periodic;
+        }
+        else if (suiteType == SuiteTypeAsString(SuiteType::Sandbox))
+        {
+            return SuiteType::Sandbox;
+        }
+        else
+        {
+            throw Exception(AZStd::string::format("Unexpected suite type: '%s'", suiteType.c_str()));
+        }
+    }
+
+    Client::SequenceReportType SequenceReportTypeFromString(const AZStd::string& type)
+    {
+        if (type == SequenceReportTypeAsString(Client::SequenceReportType::ImpactAnalysisSequence))
+        {
+            return Client::SequenceReportType::ImpactAnalysisSequence;
+        }
+        else if (type == SequenceReportTypeAsString(Client::SequenceReportType::RegularSequence))
+        {
+            return Client::SequenceReportType::RegularSequence;
+        }
+        else if (type == SequenceReportTypeAsString(Client::SequenceReportType::SafeImpactAnalysisSequence))
+        {
+            return Client::SequenceReportType::SafeImpactAnalysisSequence;
+        }
+        else if (type == SequenceReportTypeAsString(Client::SequenceReportType::SeedSequence))
+        {
+            return Client::SequenceReportType::SeedSequence;
+        }
+        else
+        {
+            throw Exception(AZStd::string::format("Unexpected sequence report type: '%s'", type.c_str()));
+        }
```

Comment:
All these functions could be simplified to a vector and a loop like this:
````
const AZStd::vector<Client::SequenceReportType> sequenceReportTypes = {{
    Client::SequenceReportType::ImpactAnalysisSequence,
    Client::SequenceReportType::RegularSequence,
    Client::SequenceReportType::SafeImpactAnalysisSequence,
    Client::SequenceReportType::SeedSequence
    }};

for (const auto& sequenceReporType : sequenceReportTypes )
{
    if (type == SequenceReportTypeAsString(sequenceReporType))
    {
        return sequenceReporType ;
    }
}

throw Exception(AZStd::string::format("Unexpected sequence report type: '%s'", type.c_str()));
````

Reply:
That seems safer, will fix 👍

Reply:
Actually, I've leave this for another PR as there's multiple funcs that might benefit from this.